pulp_container/app/viewsets.py

```diff
@@ -562,3 +562,44 @@ class ContainerNamespaceViewSet(
     endpoint_name = "pulp_container/namespaces"
     queryset = models.ContainerNamespace.objects.all()
     serializer_class = serializers.ContainerNamespaceSerializer
+    permission_classes = (AccessPolicyFromDB,)
+    queryset_filtering_required_permission = "container.view_containernamespace"
+
+    DEFAULT_ACCESS_POLICY = {
+        "statements": [
+            {
+                "action": ["list"],
+                "principal": "authenticated",
+                "effect": "allow",
+            },
+            {
+                "action": ["create"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_perms:container.add_containernamespace",
+            },
+            {
+                "action": ["retrieve"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_or_obj_perms:container.view_containernamespace",
+            },
+            {
+                "action": ["destroy"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_or_obj_perms:container.delete_containernamespace",
+            },
+        ],
+        "permissions_assignment": [
+            {
+                "function": "add_for_object_creator",
+                "parameters": None,
+                "permissions": [
+                    "container.view_containernamespace",
+                    "container.delete_containernamespace",
+                    "container.add_containerdistribution",
```

Comment:
now i am not sure, should we also add view/delete_distribution?

Reply:
Good questions. I believe you get all the required permissions of the distributions you created for free. But you could be assigned to be the "manager of all nested distributions".

Reply:
we decided to manage distribution permissions auto-assignment in the follow up prs once we get policy for the distributions.